lib/chat/db/scope/key_scope.ex

```diff
@@ -143,36 +143,34 @@ defmodule Chat.Db.Scope.KeyScope do
   end
 
   defp add_invitation_dialogs(acc_set, snap, pub_keys) do
-    dialog_keys =
+    context =
       snap
-      |> db_stream({:dialogs, 0}, {:"dialogs\0", 0})
-      |> Stream.filter(fn {_full_dilaog_key, %Dialog{a_key: a_key, b_key: b_key}} ->
-        MapSet.member?(pub_keys, a_key) or MapSet.member?(pub_keys, b_key)
-      end)
-      |> Stream.map(&just_keys/1)
-      |> Enum.to_list()
-      |> MapSet.new()
-
-    dialog_binkeys =
-      dialog_keys
-      |> Enum.map(fn {:dialogs, dialog_key} -> dialog_key end)
-      |> MapSet.new()
-
-    snap
-    |> db_stream({:dialog_message, 0, 0, 0}, {:"dialog_message\0", 0, 0, 0})
-    |> Stream.filter(fn {{:dialog_message, key, _, _}, %Message{type: type}} ->
-      type == :room_invite and MapSet.member?(dialog_binkeys, key)
-    end)
-    |> Stream.map(&just_keys/1)
-    |> union_set(dialog_keys)
-    |> union_set(acc_set)
+      |> put_invitation_dialogs()
+      |> put_checkpoints_dialogs_keys([snap, pub_keys])
+      |> put_operators_keys(snap)
+      |> put_users_invitations_dialogs_keys(snap)
+      |> put_nested_users_keys([snap, pub_keys])
+      |> put_users_invitations_dialogs_keys(snap)
+      |> put_invitations_info()
+
+    context_info = context["invitations_info"]
+
+    {
+      context_info.messages
+      |> get_messages_keys()
+      |> merge_dialogs_keys(
+        acc_set,
+        context_info.dialogs_keys
+      ),
+      full_hadshake_keys(pub_keys, context["nested_users_keys"])
+    }
```

Comment:
A glimpse  of readability :)

It was returnig a MapSet before, now a tuple. Is it ok ?

Reply:
And this is bruteforce variation.

Reply:
I change it to pass full needed keys into add_invitation_content func which is next step of backup creation

Reply:
recursive variation done